src/common/walFilter/greenplumCommon.h

```diff
@@ -0,0 +1,105 @@
+#ifndef PGBACKREST_GREENPLUMCOMMON_H
+#define PGBACKREST_GREENPLUMCOMMON_H
+
+#include <stdint.h>
+#include "postgres/interface/static.vendor.h"
+
+// Common macros and constants for different versions of Greenplum
+
+// types
+typedef uint32_t TimeLineID;
+typedef uint64_t XLogRecPtr;
+typedef uint8_t RmgrId;
+typedef uint32_t pg_crc32;
+typedef uint32_t TransactionId;
+typedef uint32_t BlockNumber;
+
+// constants
+#define XLOG_BLCKSZ 32768
+/* This flag indicates a "long" page header */
+#define XLP_LONG_HEADER             0x0002
+#define MAXIMUM_ALIGNOF 8
+#define BLCKSZ 32768
+#define XLR_MAX_BKP_BLOCKS      4
+#define XLP_FIRST_IS_CONTRECORD     0x0001
+#define XLP_FIRST_IS_OVERWRITE_CONTRECORD 0x0008
+
+#define RM_XLOG_ID                      0
+#define XLOG_SWITCH                     0x40
+#define XLOG_NOOP                       0x20
+#define MAXPGPATH       1024
+// macros
+#define MAXALIGN(LEN)           TYPEALIGN(MAXIMUM_ALIGNOF, (LEN))
+#define TYPEALIGN(ALIGNVAL,LEN)  \
+    (((uintptr_t) (LEN) + ((ALIGNVAL) - 1)) & ~((uintptr_t) ((ALIGNVAL) - 1)))
+
+#define SizeOfXLogShortPHD  MAXALIGN(sizeof(XLogPageHeaderData))
+#define SizeOfXLogLongPHD   MAXALIGN(sizeof(XLogLongPageHeaderData))
+#define XLogPageHeaderSize(hdr)     \
+    (((hdr)->xlp_info & XLP_LONG_HEADER) ? SizeOfXLogLongPHD : SizeOfXLogShortPHD)
+#define SizeOfXLogRecord    MAXALIGN(sizeof(XLogRecord))
+#define XLogRecGetData(record)  ((char*) (record) + SizeOfXLogRecord)
+#define XLR_BKP_BLOCK(iblk)     (0x08 >> (iblk))        /* iblk in 0..3 */
+#define Min(x, y)       ((x) < (y) ? (x) : (y))
+
+#define XLOG_SEG_SIZE (64 * 1024 * 1024)
+#define UINT64CONST(x) (x##UL)
+#define XLogSegmentsPerXLogId   (UINT64CONST(0x100000000) / XLOG_SEG_SIZE)
+#define XLogFromFileName(fname, tli, logSegNo)  \
+    do {                                                \
+        uint32 log;                                     \
+        uint32 seg;                                     \
+        sscanf(fname, "%08X%08X%08X", tli, &log, &seg); \
+        *logSegNo = (uint64) log * XLogSegmentsPerXLogId + seg; \
+    } while (0)
+
+#define XLogFilePath(path, tli, logSegNo)   \
+    snprintf(path, MAXPGPATH, "%08X%08X%08X", tli,              \
+             (uint32) ((logSegNo) / XLogSegmentsPerXLogId),             \
+             (uint32) ((logSegNo) % XLogSegmentsPerXLogId))
+
+// structs
+typedef struct XLogPageHeaderData
+{
+    uint16_t xlp_magic;           /* magic value for correctness checks */
```

Comment:
original definition hasn't `_t` postfix

Reply:
Does it make sense to add types from gpdb if they have an equivalent in the standard library?

Reply:
these types are not from gpdb, but from postgres

Reply:
Anyway, are there advantages to using definitions from Postgres?

Reply:
they are [already defined in pgbackrest](https://github.com/arenadata/pgbackrest/blob/0b1471b26eb72c388f934a181bb0f64e4c4aa0cc/src/postgres/interface/static.vendor.h#L36)